new datastruct workshop (#217)
NuevoFoundation/workshops: content/english/datastruct/2D Arrays/introto2darray.md

---
title: "Introduction"
draft: false
weight: 1
---
![image](../../img/2dintro.png)
<link rel="stylesheet" href="../../style.css">
<hr>

We've already learned plenty about arrays. We know that an array can hold many different types (ints, Strings, doubles, etc.), but how about an array that holds many arrays?

A 2D array has a type such as ```int[][]``` or `String[][]`, with two pairs of square brackets. The elements of a 2D array are arranged in rows and columns, and the new operator for 2D arrays specifies both the number of rows and the number of columns. For example,

```js javascript

{
    int[][] A;
    A = new int[3][4];
}

```

> This creates a 2D array of `int` that has 12 elements arranged in 3 rows and 4 columns. Although I haven't mentioned it, there are initializers for 2D arrays. For example, this statement creates the 3-by-4 array that is shown in the picture below:

Reviewer: ```suggestion
This creates a 2D array of `int` that has 12 elements arranged in 3 rows and 4 columns. There are also initializers for 2D arrays. For example, this statement creates the 3-by-4 array that is shown in the picture below:
```
